addons/mail/static/src/core/common/message.js

```diff
@@ -411,12 +411,17 @@ export class Message extends Component {
         if (ev.target.tagName === "A") {
             if (model && id) {
                 ev.preventDefault();
-                this.env.services.action.doAction({
+                await this.env.services.action.doAction({
                     type: "ir.actions.act_window",
                     res_model: model,
                     views: [[false, "form"]],
                     res_id: id,
                 });
+                if (!this.env.isSmall) {
```

Comment:
Note: in mobile this is not enough, because the chat window is originally open, what should happen is to specifically close it (if we want to see the form).

I'm not sure if that's what we want or not, it's a bit out of scope of the PR, it's already like that before.